rust/worker/src/execution/operators/metadata_filtering.rs

```diff
@@ -522,7 +523,10 @@ impl Operator<MetadataFilteringInput, MetadataFilteringOutput> for MetadataFilte
                         }
                         match record.merged_document_ref() {
                             Some(doc) => {
-                                if doc.contains(query) {
+                                // Emulate sqlite behavior. _ and % match to any character in sqlite.
+                                let normalized_query = query.replace("_", ".").replace("%", ".");
+                                let re = Regex::new(normalized_query.as_str()).unwrap();
```

Comment:
why do we need to switch to regex over contains? is that for non-ascii support?

Reply:
Yes so that the replaced `.` can match any character. We do a `LIKE %query%` in Sqlite and if query contains _ or % then sqlite matches any character

Reply:
Ah ok got it